### PR TITLE
Api reads identified unified items

### LINKED ISSUE #196: Api crashes when trying to parse results from elasticsearch
Our underlying data model has changed and it no longer reflects what we have in elasticsearch. As a result the API crashes when it tries to parse the items.
We should delete what we have in elasticsearch (once we have the pipeline for Miro working) and change the api to expose the information contained in the identified unified items

### PR DESCRIPTION
**Pull request by alicefuzier**
## What is this PR trying to achieve?
Fixes #196 
Make the API able to retrieve and show data ingested through our pipeline. Also clean some of the old unused code from the API 
## Who is this change for?
Anyone who wants to use our platform
## Have the following been considered/are they needed?

- [x] Tests?
- [x] Docs?
- [x] Spoken to the right people?

### CODE REVIEW

File: api/src/test/scala/uk/ac/wellcome/platform/api/services/ElasticSearchServiceTest.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        .hits should have size identifiedUnifiedItems.size
+    }
+  }
+
+  private def identifiedUnifiedItemWith(canonicalId: String, label: String) = {
+    IdentifiedUnifiedItem(canonicalId,
+                          UnifiedItem(identifiers = List(
+                                        SourceIdentifier(source = "Calm",
+                                                         sourceId = "AltRefNo",
+                                                         value = "calmid")),
```
Comment (alexwlchan) [inline api/src/test/scala/uk/ac/wellcome/platform/api/services/ElasticSearchServiceTest.scala]: Icky test data.

File: api/src/main/scala/uk/ac/wellcome/platform/api/Server.scala [hunk 2]
```diff
@@ -38,7 +30,8 @@ class Server extends HttpServer {
     name = "api.prefix",
     default = "/" + apiName() + "/" + apiVersion(),
     help = "API path prefix")
-
+  flag[String]("es.index", "records", "ES index name")
+  flag[String]("es.type", "item", "ES document type")
```
Comment (jtweed) [inline api/src/main/scala/uk/ac/wellcome/platform/api/Server.scala]: Have commented on #206

File: api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala [hunk 3]
```diff
@@ -11,21 +11,37 @@ import uk.ac.wellcome.platform.api.responses.{
...
+                                elasticService: ElasticSearchService)
+    extends Controller
     with SwaggerSupport {
 
   override implicit protected val swagger = ApiSwagger
 
   prefix(apiPrefix) {
+    // This is a demo endpoint for the UX team to use when prototyping
+    // item pages.
+    // TODO: Remove this endpoint.
```
Comment (jtweed) [inline api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala]: #212

File: api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticSearchService.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  def findRecords(): Future[Array[Record]] =
+    elasticClient
+      .execute {
+        search(s"$index/$itemType")
+          .matchAll()
+          // Sort so that we always have a consistent result that we can assert on
+          .sortBy(fieldSort("canonicalId.keyword"))
+          .limit(10)
+      }
+      .map { _.hits.map { Record(_) } }
```
Comment (jtweed) [inline api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticSearchService.scala]: Absolutely right. Default order TBC, but likely to be based on one of the date fields.

File: api/src/test/scala/uk/ac/wellcome/platform/api/ApiWorksTest.scala [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+          """.stripMargin
+      )
+    }
+  }
+
+  test("it should return a not found error when requesting a single work with a non existing id") {
+    server.httpGet(
+      path = "/catalogue/v0/works/non-existing-id",
+      andExpect = Status.NotFound,
+      withJsonBody = ""
```
Comment (jtweed) [inline api/src/test/scala/uk/ac/wellcome/platform/api/ApiWorksTest.scala]: Yes we should, I'm still definitely what it should look like. Will add a separate ticket.

Comment (jtweed) [inline api/src/test/scala/uk/ac/wellcome/platform/api/ApiWorksTest.scala]: Created the currently completely uninformative #213

File: api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticSearchService.scala (hunk 4 above)
Comment (alicefuzier) [inline api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticSearchService.scala]: Created #216 for it

**Pull request merged**